src/main/resources/assets/js/app/create/NewContentDialog.ts

```diff
@@ -318,11 +319,35 @@ export class NewContentDialog
         return requests;
     }
 
-    private filterContentTypes(contentTypes: ContentTypeSummaries, loginResult: LoginResult): ContentTypeSummaries {
+    private filterContentTypes(contentTypes: ContentTypeSummaries, loginResult: LoginResult): Q.Promise<ContentTypeSummaries> {
         const isContentAdmin: boolean = loginResult.isContentAdmin();
-        return contentTypes.filter(contentType => !contentType.isUnstructured() && (isContentAdmin || !contentType.isSite()));
+
+        if (isContentAdmin) {
+            return Q(this.doFilterContentTypes(contentTypes));
+        }
+
+        return ProjectHelper.isUserProjectOwnerOrEditor(loginResult).then((isOwnerOrEditor: boolean) => {
+            if (isOwnerOrEditor) {
+                return Q(this.doFilterContentTypes(contentTypes));
+            }
+
+            return Q(this.doFilterContentTypes(contentTypes, false));
```

Comment:
This can be just 

`return Q(this.doFilterContentTypes(contentTypes, isOwnerOrEditor));`